build/publisher.go

```diff
@@ -30,13 +37,22 @@ func (m *myMux) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	baseURL := m.baseURL
 	documentRoot := m.documentRoot
 
-	xffs := req.Header["X-Forwarded-For"]
-	if len(xffs) > 0 {
-		// From HAProxy's documentation:
-		// Since this header is always appended at the end of the existing header list, the server must be configured to always use the last occurrence of this header only.
-		xff := xffs[len(xffs)-1]
+	incoming_xffs := req.Header["X-Forwarded-For"]
+	merged_xffs := []string{}
+	if len(incoming_xffs) > 0 {
+		// Handle X-Forwarded-For as described by Mozilla
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/X-Forwarded-For
+		for _, header := range incoming_xffs {
+			adresses := strings.Split(header, ",")
+			for _, adress := range adresses {
+				adress := strings.TrimSpace(adress)
+				merged_xffs = append(merged_xffs, adress)
```

Comment:
We never verify that the string is a valid IP address. Not sure this is needed but worth considering.

Reply:
Not saying it's a good answer but we didn't do it before either. With that said I think it would be good to have some sort of protection since we actually parse the logs looking for addresses.

It is possible that HAproxy and/or apache adds some protection here but I know that others run this server with out a proxy aswell.

Created #11  to fix this later since I really want to the other things merged and tested.

Reply:
The fix was to easy to not include in this PR 🤦‍♂️